main/inc/global.inc.php

```diff
@@ -80,6 +80,22 @@
 // Fix bug in IIS that doesn't fill the $_SERVER['REQUEST_URI'].
 api_request_uri();
 
+// Set web proxy environment variables
+foreach ([
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/https/proxy'),
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/http/proxy'),
+             api_get_configuration_sub_value('proxy_settings/curl_setopt_array/CURLOPT_PROXY'),
+         ] as $value) {
+    if ($value && is_string($value)) {
+        foreach (['https_proxy', 'http_proxy', 'HTTPS_PROXY', 'HTTP_PROXY'] as $envVar) {
+            if (false === getenv($envVar)) {
+                putenv("$envVar=$value");
```

Comment:
Could you make sure that CURL and Guzzle use getenv() and not $_ENV?
Because of this comment: https://www.php.net/manual/en/function.putenv.php#98540, I'm a bit worried that it could not get what you put in ENV through putenv().

Reply:
\GuzzleHttp\Client::configureDefaults uses getenv().
Curl reads environment variables directly (coded in C).
[To set real environment variables, you must use putenv()](https://www.php.net/manual/en/reserved.variables.environment.php#114811)